WSL/disk-space.md

```diff
@@ -0,0 +1,207 @@
+
+```powershell
+An error occurred mounting the distribution disk, it was mounted read-only as a fallback.
+```
+
+When a distribution is started as read-only, any attempts to write to the filesystem will fail with an error like this:
+
+```bash
+$ touch file
+touch: cannot touch 'file': Read-only file system
+```
+
+To repair a disk mount error in WSL, and restore it back to a usable / writeable state again, you can use the `wsl.exe --mount` command to re-mount the disk with the following steps:
+
+1. Shutdown all WSL distributions by opening PowerShell and entering the command:
+
+    ```powershell
+    wsl.exe --shutdown
+    ```
+
+2. Open PowerShell as administrator (in an elevated command prompt) and enter the mount command, replacing `<path-to-ext4.vhdx>` with the path to the distribution's .vhdx file. For help locating this file, see [How to locate the VHD file and disk path for your Linux distribution](#how-to-locate-the-vhdx-file-and-disk-path-for-your-linux-distribution).
+
+    ```powershell
+    wsl.exe --mount <path-to-ext4.vhdx> --vhd --bare
+    ```
+
+3. Use the `wsl.exe lsblk` command from PowerShell to identify the block device name for the distribution (sd[a-z]) and then enter the following command to repair the disk (replacing `<device>` with the correct block device name, like "sdc"). The `e2fsck` command checks ext4 file systems (the type used by distributions installed with WSL) for errors and repairs them accordingly.
+
+    ```powershell
+    wsl.exe sudo e2fsck -f /dev/<device>
+    ```
+
+4. Once the repair is complete, unmount the disk in PowerShell by entering:
+
+    ```powershell
+    wsl.exe --unmount
+    ```
+
+> [!WARNING]
+> You can use the command: `sudo mount -o remount,rw /` to return a read-only distribution to a usable/writable state, but all changes will be in-memory and so will be lost when the distribution is restarted. We recommend using the steps listed above to mount and repair the disk instead.
+
+## How to locate the .vhdx file and disk path for your Linux distribution
+
+To locate the *.vhdx* file and directory path for a Linux distribution, open PowerShell and use the following script, replacing `<distribution-name>` with the actual distribution name:
+
+```powershell
+(Get-ChildItem -Path HKCU:\Software\Microsoft\Windows\CurrentVersion\Lxss | Where-Object { $_.GetValue("DistributionName") -eq '<distribution-name>' }).GetValue("BasePath") + "\ext4.vhdx"
+```
+
+The result will display a path looking something like `%LOCALAPPDATA%\Packages\<PackageFamilyName>\LocalState\<disk>.vhdx`. For example:
+
+```powershell
+C:\Users\User\AppData\Local\Packages\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc\LocalState\ext4.vhdx
+```
+
+Copy the directory path to this `ext4.vhdx` file by highlighting it and either right-clicking or selecting Ctrl+C.
```

Comment:
Let's remove this since it might not be true depending on the terminal that the user is using and its configuration

Reply:
Updated to remove the bit about highlighting/right-clicking which may be terminal-specific.